File: festival-kotlin-web/src/main/java/cn/bdqfork/web/route/response/AbstractResponseHandler.java
package cn.bdqfork.web.route.response;

import io.vertx.core.http.HttpServerResponse;

/**
 * @author bdq
 * @since 2020/2/12
 */
public abstract class AbstractResponseHandler implements ResponseHandleStrategy {

    @Override
    public void handle(HttpServerResponse httpServerResponse, Object result) throws Exception {
        if (result == null) {
            httpServerResponse.end();
        }
        doHandle(httpServerResponse, result);
    }

    protected abstract void doHandle(HttpServerResponse httpServerResponse, Object result) throws Exception;
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/response/AbstractResponseHandler.kt
package cn.bdqfork.kotlin.web.route.response

import io.vertx.ext.web.RoutingContext

/**
 * @author bdq
 * @since 2020/2/12
 */
abstract class AbstractResponseHandler : ResponseHandleStrategy {
    @Throws(Exception::class)
    override fun handle(routingContext: RoutingContext, result: Any?) {
        if (result == null) {
            routingContext.response().end()
        }
        doHandle(routingContext, result!!)
    }

    @Throws(Exception::class)
    protected abstract fun doHandle(routingContext: RoutingContext, result: Any)
}